### PR TITLE
chore: 576 test coverage: apps/web/modules/survey/list/lib

### PR DESCRIPTION
**Pull request by jakobsitory**
## What does this PR do?

Fixes [#567](https://github.com/formbricks/internal/issues/576)
Added Tests to apps/web/modules/survey/list/lib

## How should this be tested?

- let sonarQube do the heavy lifting

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [X] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

## Summary by CodeRabbit

- **Tests**
  - Added comprehensive unit tests for environment-related functions, covering scenarios such as existence checks, project ID retrieval, and error handling.
  - Introduced detailed test coverage for survey operations, including counting, retrieval, sorting, deletion, copying between environments, and error conditions.
  - Added tests for filter formatting utilities to ensure correct handling of various filter scenarios and edge cases.

### CODE REVIEW

File: apps/web/modules/survey/list/lib/survey.test.ts [hunk 1]
```diff
@@ -0,0 +1,834 @@
...
+  test("should throw DatabaseError on Prisma error", async () => {
+    const prismaError = new Prisma.PrismaClientKnownRequestError("Test Prisma Error", {
+      code: "P2001",
+      clientVersion: "test",
+      meta: {},
+    });
+    vi.mocked(prisma.survey.count).mockRejectedValue(prismaError);
+    await expect(getSurveyCount(environmentId)).rejects.toThrow(DatabaseError);
+    expect(logger.error).toHaveBeenCalledWith(prismaError, "Error getting survey count");
+  });
```
[the remark this answers is not in the document]
Comment (jakobsitory) [reply apps/web/modules/survey/list/lib/survey.test.ts:205]: Fixed in next commit

**Review by victorvhs017 — APPROVED** [review]
